### PR TITLE
[v6r15] SiteDirector enhanced logic for using empty sites

### PR DESCRIPTION
**Pull request by atsareg**
This PR is solving the problem of the sites that are not getting pilots, although not having any waiting pilots, if there are already many pilots sent to other sites. In this case empty sites still will get some pilots at a lower than usual pace

### CODE REVIEW

**Comment by andresailer** [comment]
Please also update the configuration documentation with the new option

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 1]
```diff
@@ -626,7 +634,20 @@ def getQueueSlots( self, queue ):
...
     totalSlots = self.queueSlots[queue].get( 'AvailableSlots', 0 )
+
+    # See if there are waiting pilots for this queue. If not, allow submission
+    if totalSlots and manyWaitingPilotsFlag:
+      result = pilotAgentsDB.selectPilots( {'DestinationSite':ceName,
+                                            'Queue':queueName,
+                                            'Status': WAITING_PILOT_STATUS } )
+      if result['OK']:
+        jobIDList = result['Value']
+        if len( jobIDList ) == 0:
```
Comment (chaen) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: if not jobIDList:
(pep8)

**Comment by andresailer** [comment]
I still hope you will update this documentation:
https://github.com/DIRACGrid/DIRACDocs/blob/master/source/AdministratorGuide/Configuration/ConfReference/Systems/WorkloadManagement/Agents/SiteDirector/index.rst

**Comment by fstagni** [comment]
I have updated the new doc for v6r15 (PR by this evening).

**Pull request merged**